chain/ethereum/src/adapter.rs

```diff
@@ -402,11 +402,19 @@ impl FromIterator<(BlockNumber, Address, FunctionSelector)> for EthereumCallFilt
 
 impl From<EthereumBlockFilter> for EthereumCallFilter {
     fn from(ethereum_block_filter: EthereumBlockFilter) -> Self {
+        EthereumCallFilter::from(&ethereum_block_filter)
+    }
+}
+
+impl From<&EthereumBlockFilter> for EthereumCallFilter {
```

Comment:
Generally if a `From<&T>` impl is effectively cloning `T` internally as this is doing, I prefer having only the `From<T>` impl and requiring the caller to clone if they want to use it on an `&T`, to make the potential cost of cloning explicit.

Feel free to merge after addressing this.

Reply:
It's not exact that it's cloning `T` directly, it's actually cloning only what is required for the transformation to work. I  don't see why cloning `EthereumBlockFilter ` would be necessary when we only care about some parts of the actual `EthereumBlockFilter` struct. The overprice is not quite big here, but still present. We have other occurrences of this pattern in other location like `BlockPtr::From<EthereumBlock>`.

I do understand the concern about the implicit "copy", but any `From<>` calls does some kind of allocation, so it's already implied there is some allocations cost when dealing with `From<>` calls.

Reply:
True that this can be implemented more efficiently on `&T` since it only clones the necessary bits, but it does pessimize the implementation for `T` since that now clones when previously it moved the addresses. Is `From< EthereumBlockFilter>` even used anywhere? If not we can just remove it.

Reply:
Ok, I see the point about the other version being now less efficient. I removed it and every places now uses the reference version.